crates/iota-indexer/src/store/pg_indexer_store.rs

```diff
@@ -68,6 +71,28 @@ use crate::{
     },
 };
 
+/// A cursor representing the global order position of transaction according to tx_global_order table
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct TxGlobalOrderCursor {
+    pub global_sequence_number: i64,
+    pub optimistic_sequence_number: i64,
+}
+
+impl TxGlobalOrderCursor {
+    pub fn new(global_sequence_number: i64, optimistic_sequence_number: i64) -> Self {
+        Self {
+            global_sequence_number,
+            optimistic_sequence_number,
+        }
+    }
+}
+
+impl From<(i64, i64)> for TxGlobalOrderCursor {
+    fn from((global_sequence_number, optimistic_sequence_number): (i64, i64)) -> Self {
+        Self::new(global_sequence_number, optimistic_sequence_number)
+    }
+}
```

Comment:
Nit: I think this is error prone, since it still requires the user to care for the order of the `i64` for values in the tuple, so that it correspond to the correct sequence numbers.

I would suggest just relying on the `new` constructor for this one.

Reply:
Refactored code to only use the TxGlobalOrderCursor { } construction